src/Payments/Integrations/PayPal.php

```diff
@@ -73,6 +71,17 @@ public function get_payment_method_data() {
 		return [
 			'title'       => $this->get_setting( 'title' ),
 			'description' => $this->get_setting( 'description' ),
+			'supports'    => apply_filters( '__experimental_woocommerce_blocks_payment_gateway_features_list', $this->get_supported_features(), $this->get_name() ),
```

Comment:
Wouldn't be more cleaner to have this filter inside `get_supported_features`?

Apart from that, this is still fuzzy to me, why does WCS has to declare support? how is it done in Stripe? and why can't we do it the same way it is in Stripe?

Reply:
PayPal does not support subscriptions on its own. The only way it supports it is by all the thins Subscription plugin adds on top of it to make it work with subscription payments. Without the Subscription plugin, PayPal only supports products and refunds:
https://github.com/woocommerce/woocommerce/blob/a76f9d9096b54c4a8e4d206c52ad5b3764b46d66/includes/gateways/paypal/class-wc-gateway-paypal.php#L34-L37
This is why the plugin injects the additional capabilities and since we are using the capabilities check we need a place where we can capture the capabilieties.

Reply:
> Wouldn't be more cleaner to have this filter inside get_supported_features?
I think that this is the same level of clean to have it here but I am happy to move it if you think that it will be better.

Reply:
I don't have strong feelings here, but I think in Core, the filter is applied before the return statement of a function, this means if we wanted to reach those values again in another place, you would simply call the function instead of the function + filter.
Unless you think the only place the filter should be applied is in that line alone, in which case it would make sense to leave it there, but I don't think this is the case.